src/main/scala/com/amazonaws/services/s3/AmazonS3URI.java

```diff
@@ -0,0 +1,408 @@
+/*
+ * Copyright 2014-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.services.s3;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A URI wrapper that can parse out information about an S3 URI.
+ */
+public class AmazonS3URI {
```

Comment:
I had to copy this file from the AWS SDK v1 because this functionality is not supported in v2 https://github.com/aws/aws-sdk-java-v2/issues/272. In that ticket, the suggestion is to use the v1 code. 

Are you okay with this? Should we use a different package name, like `fm.sbt.com.amazonaws.services.s3.AmazonS3URI`?